### PR TITLE
Abseil dual build and many other migrations.

### PR DESCRIPTION
**Pull request by hmaarrfk**
- [x] Needs https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/2868
- [x] Builds all platforms
- [x] Cleanup commit history by squashing it once parameters are found

Needs backporting to other branches

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
it is unclear to me what the best solution is here, should we bump the minimum requirements for OSX:

```
FAILED: src/arrow/CMakeFiles/arrow_objlib.dir/config.cc.o 
/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/_build_env/bin/x86_64-apple-darwin13.4.0-clang++ -DARROW_EXPORTING -DARROW_HAVE_RUNTIME_AVX2 -DARROW_HAVE_RUNTIME_AVX512 -DARROW_HAVE_RUNTIME_BMI2 -DARROW_HAVE_RUNTIME_SSE4_2 -DARROW_HDFS -DARROW_JEMALLOC -DARROW_JEMALLOC_INCLUDE_DIR="" -DARROW_MIMALLOC -DARROW_WITH_BACKTRACE -DARROW_WITH_BROTLI -DARROW_WITH_BZ2 -DARROW_WITH_LZ4 -DARROW_WITH_RE2 -DARROW_WITH_SNAPPY -DARROW_WITH_TIMING_TESTS -DARROW_WITH_UTF8PROC -DARROW_WITH_ZLIB -DARROW_WITH_ZSTD -DURI_STATIC_BUILD -I/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/build/src -I/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src -I/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/generated -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/thirdparty/flatbuffers/include -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/build/jemalloc_ep-prefix/src -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/build/mimalloc_ep/src/mimalloc_ep/include/mimalloc-1.7 -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/build/xsimd_ep/src/xsimd_ep-install/include -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/thirdparty/hadoop/include -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work=/usr/local/src/conda/arrow-cpp-8.0.0 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place=/usr/local/src/conda-prefix -Qunused-arguments -fcolor-diagnostics -O3 -DNDEBUG  -Wall -Wno-unknown-warning-option -Wno-pass-failed  -O3 -DNDEBUG -isysroot /Applications/Xcode_12.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -mmacosx-version-min=10.9 -fPIC -std=c++17 -MD -MT src/arrow/CMakeFiles/arrow_objlib.dir/config.cc.o -MF src/arrow/CMakeFiles/arrow_objlib.dir/config.cc.o.d -o src/arrow/CMakeFiles/arrow_objlib.dir/config.cc.o -c /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/arrow/config.cc
In file included from /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/arrow/config.cc:24:
In file included from /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/arrow/vendored/datetime.h:20:
/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/arrow/vendored/datetime/date.h:1069:22: error: 'uncaught_exceptions' is unavailable: introduced in macOS 10.12 - see https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
                std::uncaught_exceptions() == 0 &&
                     ^
/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/_build_env/bin/../include/c++/v1/exception:134:63: note: 'uncaught_exceptions' has been explicitly marked unavailable here
_LIBCPP_FUNC_VIS _LIBCPP_AVAILABILITY_UNCAUGHT_EXCEPTIONS int uncaught_exceptions() _NOEXCEPT;
                                                              ^
1 error generated.
```

Should we do it selectively? I would rather do it for all OSX builds.

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
@xhochy I believe that htis is the last update to ensure we finish the grpc/abseil migration.

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,3 +1,10 @@
+MACOSX_SDK_VERSION:        # [osx and x86_64]
```
Comment (xhochy) [inline recipe/conda_build_config.yaml:1]: Where is the SDK version requirement coming from?

**Comment by xhochy** [comment]
> 



> it is unclear to me what the best solution is here, should we bump the minimum requirements for OSX:
> 
> ```
> FAILED: src/arrow/CMakeFiles/arrow_objlib.dir/config.cc.o 
> /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/_build_env/bin/x86_64-apple-darwin13.4.0-clang++ -DARROW_EXPORTING -DARROW_HAVE_RUNTIME_AVX2 -DARROW_HAVE_RUNTIME_AVX512 -DARROW_HAVE_RUNTIME_BMI2 -DARROW_HAVE_RUNTIME_SSE4_2 -DARROW_HDFS -DARROW_JEMALLOC -DARROW_JEMALLOC_INCLUDE_DIR="" -DARROW_MIMALLOC -DARROW_WITH_BACKTRACE -DARROW_WITH_BROTLI -DARROW_WITH_BZ2 -DARROW_WITH_LZ4 -DARROW_WITH_RE2 -DARROW_WITH_SNAPPY -DARROW_WITH_TIMING_TESTS -DARROW_WITH_UTF8PROC -DARROW_WITH_ZLIB -DARROW_WITH_ZSTD -DURI_STATIC_BUILD -I/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/build/src -I/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src -I/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/generated -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/thirdparty/flatbuffers/include -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/build/jemalloc_ep-prefix/src -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/build/mimalloc_ep/src/mimalloc_ep/include/mimalloc-1.7 -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/build/xsimd_ep/src/xsimd_ep-install/include -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/thirdparty/hadoop/include -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work=/usr/local/src/conda/arrow-cpp-8.0.0 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place=/usr/local/src/conda-prefix -Qunused-arguments -fcolor-diagnostics -O3 -DNDEBUG  -Wall -Wno-unknown-warning-option -Wno-pass-failed  -O3 -DNDEBUG -isysroot /Applications/Xcode_12.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -mmacosx-version-min=10.9 -fPIC -std=c++17 -MD -MT src/arrow/CMakeFiles/arrow_objlib.dir/config.cc.o -MF src/arrow/CMakeFiles/arrow_objlib.dir/config.cc.o.d -o src/arrow/CMakeFiles/arrow_objlib.dir/config.cc.o -c /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/arrow/config.cc
> In file included from /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/arrow/config.cc:24:
> In file included from /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/arrow/vendored/datetime.h:20:
> /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/arrow/vendored/datetime/date.h:1069:22: error: 'uncaught_exceptions' is unavailable: introduced in macOS 10.12 - see https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
>                 std::uncaught_exceptions() == 0 &&
>                      ^
> /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/_build_env/bin/../include/c++/v1/exception:134:63: note: 'uncaught_exceptions' has been explicitly marked unavailable here
> _LIBCPP_FUNC_VIS _LIBCPP_AVAILABILITY_UNCAUGHT_EXCEPTIONS int uncaught_exceptions() _NOEXCEPT;
>                                                               ^
> 1 error generated.
> ```
> 
> Should we do it selectively? I would rather do it for all OSX builds.

I'm confused why this wasn't an issue before.

**Comment by hmaarrfk** [comment]
> > 
> 
> > it is unclear to me what the best solution is here, should we bump the minimum requirements for OSX:
> > ```
> > FAILED: src/arrow/CMakeFiles/arrow_objlib.dir/config.cc.o 
> > /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/_build_env/bin/x86_64-apple-darwin13.4.0-clang++ -DARROW_EXPORTING -DARROW_HAVE_RUNTIME_AVX2 -DARROW_HAVE_RUNTIME_AVX512 -DARROW_HAVE_RUNTIME_BMI2 -DARROW_HAVE_RUNTIME_SSE4_2 -DARROW_HDFS -DARROW_JEMALLOC -DARROW_JEMALLOC_INCLUDE_DIR="" -DARROW_MIMALLOC -DARROW_WITH_BACKTRACE -DARROW_WITH_BROTLI -DARROW_WITH_BZ2 -DARROW_WITH_LZ4 -DARROW_WITH_RE2 -DARROW_WITH_SNAPPY -DARROW_WITH_TIMING_TESTS -DARROW_WITH_UTF8PROC -DARROW_WITH_ZLIB -DARROW_WITH_ZSTD -DURI_STATIC_BUILD -I/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/build/src -I/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src -I/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/generated -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/thirdparty/flatbuffers/include -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/build/jemalloc_ep-prefix/src -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/build/mimalloc_ep/src/mimalloc_ep/include/mimalloc-1.7 -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/build/xsimd_ep/src/xsimd_ep-install/include -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/thirdparty/hadoop/include -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work=/usr/local/src/conda/arrow-cpp-8.0.0 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place=/usr/local/src/conda-prefix -Qunused-arguments -fcolor-diagnostics -O3 -DNDEBUG  -Wall -Wno-unknown-warning-option -Wno-pass-failed  -O3 -DNDEBUG -isysroot /Applications/Xcode_12.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -mmacosx-version-min=10.9 -fPIC -std=c++17 -MD -MT src/arrow/CMakeFiles/arrow_objlib.dir/config.cc.o -MF src/arrow/CMakeFiles/arrow_objlib.dir/config.cc.o.d -o src/arrow/CMakeFiles/arrow_objlib.dir/config.cc.o -c /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/arrow/config.cc
> > In file included from /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/arrow/config.cc:24:
> > In file included from /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/arrow/vendored/datetime.h:20:
> > /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/work/cpp/src/arrow/vendored/datetime/date.h:1069:22: error: 'uncaught_exceptions' is unavailable: introduced in macOS 10.12 - see https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
> >                 std::uncaught_exceptions() == 0 &&
> >                      ^
> > /Users/runner/miniforge3/conda-bld/arrow-cpp-ext_1652651826412/_build_env/bin/../include/c++/v1/exception:134:63: note: 'uncaught_exceptions' has been explicitly marked unavailable here
> > _LIBCPP_FUNC_VIS _LIBCPP_AVAILABILITY_UNCAUGHT_EXCEPTIONS int uncaught_exceptions() _NOEXCEPT;
> >                                                               ^
> > 1 error generated.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Should we do it selectively? I would rather do it for all OSX builds.
> 
> I'm confused why this wasn't an issue before.

I'm going to venture and say that maybe the osx compiler upgrade has something to do with it

**Comment by hmaarrfk** [comment]
I'm going to trigger a build without rebasing for us to study.

**Comment by hmaarrfk** [comment]
@xhochy i guess the new OSX is necessary for the new abseil. I don't know how to bump it just for for that version (in terms of syntax that is).

**Comment by hmaarrfk** [comment]
We could, if you wanted to, merge without a bump in the osx SDK, then follow-up with a bump to have a build for osx with the old abseil for older deployment targets.

**Comment by xhochy** [comment]
> We could, if you wanted to, merge without a bump in the osx SDK, then follow-up with a bump to have a build for osx with the old abseil for older deployment targets.

No, too much of a hassle.

I'll backport the changes here to all other branches. Thank you for taking care of this!

**Comment by hmaarrfk** [comment]
Thanks. Can you take care of merging the migration closing too?

**Comment by jakirkham** [comment]
It looks like the build number didn't get bumped here. Mentioning as it looks like some of the `pyarrow` packages didn't get rebuilt (like [this one]( https://anaconda.org/conda-forge/pyarrow/files/modal/info/6286a0dc302e80ce8ca94f31 )). Even though this shouldn't affect `pyarrow`, we pin to [`arrow-cpp` exactly]( https://github.com/conda-forge/arrow-cpp-feedstock/blob/6838cae244751a49ee1458a09d115d1d12208c73/recipe/meta.yaml#L190 ). As a result `arrow-cpp`'s before the migration are pulled in (like [this one]( https://anaconda.org/conda-forge/arrow-cpp/files/modal/info/628bb73f9501cf2a1dc49671 )) with the old pinning. Went ahead and just added this change in commit ( https://github.com/conda-forge/arrow-cpp-feedstock/commit/1032f3d6a8373223db5ebc7265ffc9ebd4974838 ) to get these rebuilt. Hope that's ok

**Comment by hmaarrfk** [comment]
thank you

**Comment by hmaarrfk** [comment]
did you want to mark the other packages as broken?

**Pull request merged**